spec/models/registered_user_spec.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe RegisteredUser, type: :model do
+  it { is_expected.to respond_to(:system_admin) }
+  it { is_expected.to respond_to(:email) }
+  it { is_expected.to respond_to(:first_name) }
+  it { is_expected.to respond_to(:last_name) }
+  it { is_expected.to respond_to(:grant_creator) }
+  it { is_expected.to respond_to(:era_commons) }
+
+  let(:user)       { FactoryBot.build(:registered_user) }
+  let(:other_user) { create(:registered_user, era_commons: Faker::Lorem.characters(number: 10)) }
+
+  describe '#initiations' do
+    it 'sets default of system_admin boolean' do
+      expect(user.system_admin).to be(false)
+    end
+
+    it 'sets default of grant_creator boolean' do
+      expect(user.grant_creator).to be(false)
+    end
+
+  end
+
+  describe '#validations' do
+    it 'validates presence of email' do
+      user.email = nil
+      expect(user).not_to be_valid
+      expect(user.errors).to include :email
+    end
+
+    it 'checks for restricted domains' do
```

Comment:
I would add test for each restricted domain. You can user shared spec here, with just setting email for each restricted domain to save repeated code.